lib/ansible/plugins/action/synchronize.py

```diff
@@ -155,6 +155,11 @@ def run(self, tmp=None, task_vars=None):
         if task_vars is None:
             task_vars = dict()
 
+        # We make a copy of the args here because we may fail and be asked to
+        # retry. If that happens we don't want to pass the munged args through
+        # to our next invocation. Munged args are single use only.
+        self._task._tmp_args = self._task.args.copy()
```

Comment:
Yeah, this is a quick fix that I understand will work.  I'm not certain whether changing retries to create a new action plugin on every iteration will have unwanted (or just buggy) side-effects.  I'd merge this and cherry-pick to 2.2.  Then if we do decide that retries can be changed, we can make that change in devel only.